src/modules/generators/checkerboard.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2013 The Noise-rs Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num::{Float, One};
+use math;
+use Module;
+use math::{Point2, Point3};
+
+#[derive(Clone,Copy,Debug)]
+pub struct Checkerboard {
+    size: usize,
+}
+
+impl Checkerboard {
+    pub fn new(s: usize) -> Checkerboard {
+        Checkerboard { size: 1 << s }
+    }
+}
+
+fn fastfloor<T: Float>(x: T) -> usize {
+    if x > math::cast::<_, T>(0) {
+        math::cast::<_, usize>(x)
+    } else {
+        math::cast::<_, usize>(x - T::one())
+    }
+}
+
+impl<T: Float> Module<Point2<T>> for Checkerboard {
+    type Output = T;
+
+    fn get(&self, point: Point2<T>) -> Self::Output {
+        let result = point.iter()
+            .map(|&a| fastfloor(a))
+            .fold(0, |a, b| (a & self.size) ^ (b & self.size));
```

Comment:
I'm keeping the implementation the same between the different PointNs so that at some later point when we have a higher Point type, they can be made generic over Point instead of Point2/3/N.